pkg/objstore/azure/helpers.go

```diff
@@ -0,0 +1,71 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"strings"
+
+	blob "github.com/Azure/azure-storage-blob-go/2018-03-28/azblob"
+)
+
+var (
+	blobFormatString = `https://%s.blob.core.windows.net`
+)
+
+type blobReadCloser struct {
```

Comment:
I think we can kill this struct and use just `ioutil.NopCloser(reader)`

Reply:
No, unfortunately not :( I tried that. It breaks `bucket list` command as it want's to access nil value. I know it's a bit hacky but I haven't found better way to handle it.

Reply:
hm.. don't get this. Why we cannot use `return ioutil.NopCloser(bytes.NewReader(destBuffer))`